### PR TITLE
add KeepFields

### PR DESCRIPTION
**Pull request by bmizerany**

### CODE REVIEW

File: option.go [hunk 1]
```diff
@@ -142,6 +139,39 @@ func ZeroFields[T any](fields ...string) Option {
...
+				fv0 := e0.FieldByName(name)
+				fv1 := e1.FieldByName(name)
+				fv1.Set(fv0)
+			}
+		}
+		return v1
+	})
+}
+
+func checkFieldsExist[T any](fields ...string) {
```
Comment (kr) [inline option.go]: Nit: what do you think about []string here?

File: go.mod [hunk 2]
```diff
@@ -6,3 +6,5 @@ require (
 	github.com/pkg/diff v0.0.0-20210226163009-20ebb0f2a09e
 	github.com/rogpeppe/go-internal v1.8.1
 )
+
+require golang.org/x/exp v0.0.0-20220218215828-6cf2b201936e // indirect
```
Comment (kr) [inline go.mod]: Does go mod tidy get rid of the "indirect" comment?

File: option.go [hunk 3]
```diff
@@ -142,6 +139,39 @@ func ZeroFields[T any](fields ...string) Option {
 	})
 }
 
+// KeepFields transforms a value of struct type T. It makes a copy of its input
```
Comment (kr) [inline option.go]: Nit: missing comma after "input".

File: option.go (hunk 1 above)
Comment (bmizerany) [reply option.go]: Down with that. Will update.

File: go.mod (hunk 2 above)
Comment (bmizerany) [reply go.mod]: Let's see. :)

File: option.go (hunk 3 above)
Comment (bmizerany) [reply option.go]: Noted.

**Review by kr — APPROVED** [review]

**Pull request merged**